### PR TITLE
feat(influxdb): Backport modules to 1.7.x

### PR DESCRIPTION
**Pull request by ayang64**
This PR will backport the modules work done in 1.8 to the 1.7 branch.

This involves removing Gopkg.* files, adding go.mod and go.sum files, and updating the supporting CI/CD configuration and scripts.

Closes: #16901
Closes: #17057

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

**Comment by sylr** [comment]
I don't believe this should be backported to 1.7 line.

According to semver this kind of change falls into a MINOR change, not a PATCH one.

**Comment by ayang64** [comment]
> I don't believe this should be backported to 1.7 line.
> 
> According to semver this kind of change falls into a MINOR change, not a PATCH one.

so we have a 1.7 branch that the next minor release will likely be based on.  1.7 is the tip of all 1.7.x development so if we release, for example, 1.7.11, we'd like this modules work to land there.

does this make sense or am i missing something?

File: go.mod [hunk 1]
```diff
@@ -3,108 +3,54 @@ module github.com/influxdata/influxdb
 go 1.12
```
Comment (dgnorton) [inline go.mod:3]: Does this need to be updated?

Comment (ayang64) [reply go.mod:3]: d'oh -- yes, it does.

File: Dockerfile_build_ubuntu64_go1.12 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    python \
+    python-boto \
+    asciidoc \
+    xmlto \
+    docbook-xsl
+
+RUN gem install fpm
+
+# setup environment
+ENV GO_VERSION  1.12
```
Comment (dgnorton) [inline Dockerfile_build_ubuntu64_go1.12:26]: Does this need to be updated?

Comment (dgnorton) [reply Dockerfile_build_ubuntu64_go1.12:26]: Actually, the name of this file is `Dockerfile_build_ubuntu64_go1.12 ` so I'm not sure if it should be changed here or a new file created??

**Comment by ayang64** [comment]
> @ayang64 Looks like this backport introduces `Dockerfile_build_ubuntu64_go1.12 `. Do we need it in 1.7.x?

I don't think so.  The original author isn't around but as far as I can tell, it was there to test go 1.12 before it was released.

I'll delete the file and resubmit the PR.

**Comment by ayang64** [comment]
Actually -- if it is okay with you, what I'd like to do is leave it in this branch.

The docker file exists in our `master-1.x` branch too.  I'd like to open a ticket to remove it there and backport the fix to 1.7.x.

Does that sound reasonable?

**Comment by dgnorton** [comment]
I'd say just remove it from this branch so we don't have to bother with a backport to remove it.

**Pull request merged**